wcivf/apps/elections/tests/test_election_views.py

```diff
@@ -39,3 +40,45 @@ def test_election_detail_view(self):
         self.assertEqual(response.status_code, 200)
         self.assertTemplateUsed(response, "elections/election_view.html")
         self.assertContains(response, self.election.nice_election_name)
+
+class ElectionPostViewTests(TestCase):
+    def setUp(self):
+        self.election = ElectionFactory(
+            name="Adur local election",
+            election_date="2021-05-06",
+            slug="local.adur.churchill.2021-05-06",
+        )
+        self.post = PostFactory()
+        self.post_election = PostElectionFactory(election=self.election, post=self.post)
+        people = [PersonFactory() for p in range(5)]
+        for person in people:
+                PersonPostFactory(
+                post_election=self.post_election,
+                election=self.election,
+                post=self.post,
+                person=person,
+            )
+
+    def test_zero_candidates(self):
+        response = self.client.get(
+        self.election.get_absolute_url(), follow=True
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "elections/election_view.html")
+        # I need to figure out how to test the page_title on the line below
```

Comment:
@michaeljcollinsuk I couldn't figure out how to test the `page_title`; A simple response + str hasn't worked here.

Reply:
I think the reason that the assert commented out on line 69 fails is because your test is checking the wrong view - rather than the `ElectionView` it should be the `PostView` that the test needs to hit. As the template rendered by the `ElectionView` doesn't include `_post_meta_title.html `. And also it is in the `PostView` where the `people` variable that is used is set https://github.com/DemocracyClub/WhoCanIVoteFor/blob/fd79f1ad27645c0e806484f3a9ebc8884389b6c9/wcivf/apps/elections/views/election_views.py#L106

So you should be able to change line 63 to:
```
response = self.client.get(
    self.post_election.get_absolute_url(), follow=True
)
```

You will also then need to update the assert on line 67 to check that the `post_view.html` is used.

Reply:
Thanks for catching this. 👍